sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPSensor.java

```diff
@@ -233,28 +241,74 @@ private void saveParsingIssue(SensorContext context, RecognitionException e, Inp
     }
   }
 
-  private void saveIssues(SensorContext context, List<Issue> issues, InputFile inputFile) {
-    for (Issue phpIssue : issues) {
-      RuleKey ruleKey = checks.ruleKeyFor(phpIssue.check());
+  private void saveIssues(SensorContext context, List<CheckIssue> issues, InputFile inputFile) {
+    for (CheckIssue issue : issues) {
+      RuleKey ruleKey = checks.ruleKeyFor(issue.check());
+      NewIssue newIssue = context.newIssue()
+        .forRule(ruleKey)
+        .gap(issue.cost());
 
-      NewIssue issue = context.newIssue();
+      if (issue instanceof PHPIssue) {
+        // todo: this block should be removed as PHPIssue's usages will be removed
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)